azkaban-common/src/main/java/azkaban/container/models/AzKubernetesV1Service.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2020 LinkedIn Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package azkaban.container.models;
+
+import com.google.common.base.Charsets;
+import io.kubernetes.client.openapi.models.V1Service;
+import io.kubernetes.client.util.Yaml;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Map;
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.lang.text.StrSubstitutor;
+
+/**
+ * A class which creates the {@link V1Service} object based on the template and substituting values.
+ * Each instance of the class is unique to a template which will packages in the jar as a resource.
+ * After the instantiation, this class object can be reused to create different service objects.
+ */
+public class AzKubernetesV1Service {
+
+  private final String template;
+
+  /**
+   * Constructor for instantiation of this class
+   *
+   * @param templateName name of the resource within the package
+   * @throws IOException if the resource file is not readable
+   */
+  public AzKubernetesV1Service(String templateName) throws IOException {
+    try (final InputStream is = this.getClass().getResourceAsStream(templateName)) {
+      this.template = IOUtils.toString(is, Charsets.UTF_8).trim();
+    }
+  }
+
+  /**
+   * @param valuesMap Map containing the key-value pairs used for substitution in the template
+   * @return {@link V1Service} object by replacing the values from the valuesMap within the template
+   * @throws IOException if the object can't be constructed from the substituted YAML String
+   */
+  public V1Service getServiceObject(Map<String, String> valuesMap) throws IOException {
+    StrSubstitutor strSubstitutor = new StrSubstitutor(valuesMap);
```

Comment:
Does V1Service provide a builder that we could use instead of string substitution?
If there is a reason for not using it please mention that in the comments.

Reply:
We still have to end up using string substitution for the creation of annotation value which is the major chunk for the spec, hence it is better to use a template than builder itself.